ls1intum/Orion
# [Enhancement] Remove Hardcoded References to Master

### Motivation and Context
Since https://github.com/ls1intum/Artemis/pull/2844, Artemis supports any default branch. Orion currently has several hardcoded references to master that will not work with this change.

### Description
Instead of always pushing/pulling from/to master, Orion will now push/pull from the remote tracking branch of the current branch. If no such branch can be found, an error is show to the user. If multiple remote branches are present, a warning is shown to inform the user the changes might not have pushed to the main branch and therefore might not get graded.

### Steps for Testing
1.  Install the release as described in [the readme](https://github.com/ls1intum/Orion/blob/master/README.md#testing-of-pull-requests)
2.  Open a programming exercise as a student (make sure to use one where you have access to the git server)
3.  Test pushing (submitting) and pulling (automatically on startup) with various branch configurations:
    - Normal configuration with 1 branch
    - Create a local branch without remote branch (should give an error)
    - Create a different remote branch (should give a warning)
    - Switch to a different branch with remote tracking branch (should work, but give warning)
4. Also test an existing exercise to make sure no regression happened
5. (Optional) Create/Locate an exercise with the default branch not named master and try it. Theoretically the code does not mention "master" anymore, so it should not behave any different
